### ISSUE TITLE
panic: Meta node panicked at `unwrap`

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#6683]
### Describe the bug

Meta node panicked 

```
thread 'risingwave-main' panicked at 'called `Result::unwrap()` on an `Err` value: Internal(grpc request error: status: Unknown, message: "transport error", details: [], metadata: MetadataMap { headers: {} }
```

at

https://github.com/risingwavelabs/risingwave/blob/2d5546ab7506e58d7fd025ff65f860576c5ba45f/src/meta/src/barrier/mod.rs#L599-L602

The `Err` is caused by `MetaStoreError::Internal`

```
   0: anyhow::error::<impl anyhow::Error>::new
             at /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.66/src/error.rs:36:25
   1: risingwave_meta::storage::etcd_meta_store::<impl core::convert::From<etcd_client::error::Error> for risingwave_meta::storage::meta_store::MetaStoreError>::from
             at ./src/meta/src/storage/etcd_meta_store.rs:28:34
   2: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/result.rs:2108:27
   3: <risingwave_meta::storage::etcd_meta_store::EtcdMetaStore as risingwave_meta::storage::meta_store::MetaStore>::put_cf::{{closure}}
             at ./src/meta/src/storage/etcd_meta_store.rs:192:9
```

Full logs: [risingwave-logs-dec1.zip](https://github.com/risingwavelabs/risingwave/files/10129248/risingwave-logs-dec1.zip)

### To Reproduce

Run NexMark source and

```
create materialized view mv_q6s as
  SELECT
    MAX(B.price) AS final,
    A.id,
    A.seller
  FROM
    auction AS A,
    bid AS B
  WHERE
    A.id = B.auction
    and B.date_time between A.date_time and A.expires
  GROUP BY
    A.id,
    A.seller
```

### Additional context

Full logs: [risingwave-logs-dec1.zip](https://github.com/risingwavelabs/risingwave/files/10129248/risingwave-logs-dec1.zip)

**Comment by yezizp2012** [comment]
To fix similar problem in #4919, We only retry on etcd when it return `Unavailable`. In this case, the error code is `Unknown`, should we also retry on this? 🤔 Cc @fuyufjh

**Comment by fuyufjh** [comment]
> To fix similar problem in #4919, We only retry on etcd when it return `Unavailable`. In this case, the error code is `Unknown`, should we also retry on this? 🤔 Cc @fuyufjh

```
status: Unknown, message: "transport error"
```

The message said the cause is `transport error`, so I think Yes at least for this case.
